core/Db/TransactionLevel.php

```diff
@@ -54,7 +59,13 @@ public function setUncommitted()
             $this->db->query('SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED');
             $this->statusBackup = $backup;
 
-            Option::set(self::TEST_OPTION_NAME, '1'); // try setting something w/ the new transaction isolation level
+            if ($this->db->supportsUncommitted === null) {
```

Comment:
This will currently always be true, as the method already returns in the beginning